### ISSUE TITLE
Thumbnail handling

### ISSUE THREAD

**Issue by gianlucadelgobbo** [gianlucadelgobbo/avnode#71]
there was a function that accordingly to the config was lookig for a size format if there wasnt he create it if there isnt the image was taking the default image.

**Comment by gianlucadelgobbo** [comment]
Almost done but the crop callback arrives when the image is still not available, solution?

**Comment by alappe** [comment]
From a quick glance, the solution isn't that easy.

Since the imagemagick functions are are asynchronous, the `getFileFormat` function has to be async as well. Changing that is simple, but the usage seems to be inline in jade, and there we cannot use the resulting async function.

So one way would be to get the metadata and do cropping on upload, not on display.
The other way could be prepare all data such as this before the view is rendered, which would make the first view slow (since images will need to be resized).

**Comment by gianlucadelgobbo** [comment]
yes, serverside could be very slow and we need to put a lot of code to make this check on every query in node.

Another solution could be a js client side for this very rare case
`document.addEventListener('DOMContentLoaded', function () {
	var img = document.getElementById("img-#{result.permalink}");
	if (!img.complete || typeof img.naturalWidth !== "undefined" && img.naturalWidth === 0) {
		img.src += "?" + new Date().getTime();
		console.log("reload");
	}
}, false);`

**Comment by gianlucadelgobbo** [comment]
Formats are here, in every item of the sections
config/default.json
```
      "thumbnails": {
        "detail":{"w":1140,"h":641,"default":"event.jpg"},
        "list":{"w":263,"h":148,"default":"event.jpg"}
      }
```

**Comment by stefanisak** [comment]
IMHO we should store the original file on upload and resize it later based on configuration. Resizing during upload solves the performance/async issue, but makes it hard to adjust the layout once different dimensions are required.

I'll try to find a solution based on your configs and @alappe suggests and come back here.

**Comment by gianlucadelgobbo** [comment]
For sure we do it in upload, but this is just for safety reason, noo?
